Doppler.Sap.Job/DopplerCurrencyService/DopplerCurrencyService.cs

```diff
@@ -39,18 +39,18 @@ public async Task<IList<CurrencyResponse>> GetCurrencyByCode()
             
             foreach (var currencyCode in _dopplerCurrencySettings.CurrencyCodeList)
             {
-                var uri = new Uri(_dopplerCurrencySettings.Url + $"{currencyCode}/{cstTime.Year}-{cstTime.Month}-{cstTime.Day}");
+                try { 
+                    var uri = new Uri(_dopplerCurrencySettings.Url + $"{currencyCode}/{cstTime.Year}-{cstTime.Month}-{cstTime.Day}");
+                    
+                    _logger.LogInformation($"Building http request with url {uri}"); 
+                    var httpRequest = new HttpRequestMessage 
+                    {
+                        RequestUri = uri, 
+                        Method = new HttpMethod("GET")
 
-                _logger.LogInformation($"Building http request with url {uri}");
-                var httpRequest = new HttpRequestMessage
-                {
-                    RequestUri = uri,
-                    Method = new HttpMethod("GET")
-                };
-
-                try
```

Comment:
In what scenario does it throw an exception before the request?

Reply:
For example for problems in the appsettings.json with the config _dopplerCurrencySettings.Url